### PR TITLE
dm-master: fix DM-master can not exit when starting timeout

### LINKED ISSUE #507: HA: DM-master can not exit when starting timeout
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

    1. start a DM-master instance but with more than one instances in `initial-cluster `
    2. wait for DM-master to exit after the default timeout (`1min`)

2. What did you expect to see?

    the process of DM-master exits.

3. What did you see instead?

    the process of DM-master keeps running, but output some log repeatedly.

```
[2020/03/02 20:11:24.867 +08:00] [INFO] [etcd.go:360] ["closing etcd server"] [component="embed etcd"] [name=dm-master-1] [data-dir=default.dm-master] [advertise-peer-urls="[http://127.0.0.1:8291]"] [advertise-client-urls="[http://127.0.0.1:8261]"]
[2020/03/02 20:11:29.012 +08:00] [WARN] [grpclog.go:60] ["grpc: addrConn.createTransport failed to connect to {127.0.0.1:8261 0  <nil>}: didn't receive server preface in time. Reconnecting..."] [component="embed etcd"]
[2020/03/02 20:11:52.377 +08:00] [WARN] [grpclog.go:60] ["grpc: addrConn.createTransport failed to connect to {127.0.0.1:8261 0  <nil>}: didn't receive server preface in time. Reconnecting..."] [component="embed etcd"]
[2020/03/02 20:12:17.961 +08:00] [WARN] [grpclog.go:60] ["grpc: addrConn.createTransport failed to connect to {127.0.0.1:8261 0  <nil>}: didn't receive server preface in time. Reconnecting..."] [component="embed etcd"]
[2020/03/02 20:12:47.167 +08:00] [WARN] [grpclog.go:60] ["grpc: addrConn.createTransport failed to connect to {127.0.0.1:8261 0  <nil>}: didn't receive server preface in time. Reconnecting..."] [component="embed etcd"]
[2020/03/02 20:13:31.872 +08:00] [WARN] [grpclog.go:60] ["grpc: addrConn.createTransport failed to connect to {127.0.0.1:8261 0  <nil>}: didn't receive server preface in time. Reconnecting..."] [component="embed etcd"]
[2020/03/02 20:14:40.563 +08:00] [WARN] [grpclog.go:60] ["grpc: addrConn.createTransport failed to connect to {127.0.0.1:8261 0  <nil>}: didn't receive server preface in time. Reconnecting..."] [component="embed etcd"]
[2020/03/02 20:16:41.306 +08:00] [WARN] [grpclog.go:60] ["grpc: addrConn.createTransport failed to connect to {127.0.0.1:8261 0  <nil>}: didn't receive server preface in time. Reconnecting..."] [component="embed etcd"]
```

4. Versions of the cluster

     the current latest version with https://github.com/pingcap/dm/commit/f4c1a448604a7fbf72f3dcb0426c94e195225e3c.

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

fix #507 

### What is changed and how it works?

see comments in _dm/master/etcd.go_ from L61 to L68.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@WangXiangUSTC @lichunzhu PTAL

File: dm/master/etcd.go [hunk 1]
```diff
@@ -57,9 +57,16 @@ func startEtcd(etcdCfg *embed.Config,
...
+		// https://github.com/etcd-io/etcd/blob/3cf2f69b5738fb702ba1a935590f36b52b18979b/embed/serve.go#L92
+		// then `e.Close` will block in
+		// https://github.com/etcd-io/etcd/blob/3cf2f69b5738fb702ba1a935590f36b52b18979b/embed/etcd.go#L377
+		// because `close(sctx.serversC)` has not been called in
+		// https://github.com/etcd-io/etcd/blob/3cf2f69b5738fb702ba1a935590f36b52b18979b/embed/serve.go#L200.
+		// so for `ReadyNotify` timeout, we choose to only call `e.Server.Stop()` now,
+		// and we should exit the DM-master process after returned with error from this function.
 		e.Server.Stop()
-		e.Close()
 		return nil, terror.ErrMasterStartEmbedEtcdFail.Generatef("start embed etcd timeout %v", etcdStartTimeout)
```
Comment (lichunzhu) [inline dm/master/etcd.go]:
```suggestion
		return nil, terror.ErrMasterStartEmbedEtcdFail.Generatef("start embed etcd timeout %v", startTimeout)
```

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**